### ISSUE TITLE
Publishing dart-3.1.2 failied

### ISSUE THREAD

**Issue by mtrezza** [parse-community/Parse-SDK-Flutter#771]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/main/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest version of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse Flutter SDK](https://github.com/parse-community/Parse-SDK-Flutter/releases). 

### Issue Description

Publishing failed with:

```
Package validation found the following error:
* Your pubspec.yaml must not override non-dev dependencies.
  This ensures you test your package against the same versions of its dependencies
  that users will have when they use it.
Sorry, your package is missing a requirement and can't be published yet.
For more information, see: https://dart.dev/tools/pub/cmd/pub-lish.
Dry Run Failed, skip real publishing.
```

### Steps to reproduce
See https://github.com/parse-community/Parse-SDK-Flutter/runs/7231328869?check_suite_focus=true

### Actual Outcome
Publishing fails

### Expected Outcome
Publishing succeeds

### Environment
dart 3.1.2

**Comment by mtrezza** [comment]
cc @RodrigoSMarques

**Comment by RodrigoSMarques** [comment]
> cc @RodrigoSMarques

New bugs. They occurred in the sample application. I'll have to analyze it calmly and maybe change the action.

**Comment by RodrigoSMarques** [comment]
@mtrezza 
Fix with PR #772

**Comment by mtrezza** [comment]
Closing via #772
